### PR TITLE
PR: Mutation to Angular

### PR DESCRIPTION
**Pull request by AndreyAttr**
Changes:

1. Reactive form approach component for mutation(hasn't finished yet)
2. Create generic-approached EnumHelper from non-generic GeneHelper

### CODE REVIEW

File: ClientWeb/Evolution/AppState/app-state.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ const defaultState: State = {
     mutationEnabled: false,
     selectedGenes: new GeneStorage(),      //[],
     availableGenes: new GeneStorage(
-        new Gene('HFE', [new Site(SiteEnum.A), new Site(SiteEnum.C), new Site(SiteEnum.G), new Site(SiteEnum.C), new Site(SiteEnum.U), new Site(SiteEnum.C)], 'Excess of iron in the body'),  //Surplus
+        new Gene('HFE', [new Site(SiteEnum.A), new Site(SiteEnum.C), new Site(SiteEnum.G, true), new Site(SiteEnum.C), new Site(SiteEnum.U), new Site(SiteEnum.C)], 'Excess of iron in the body'),  //Surplus
```
Comment (tapy-lappy) [inline ClientWeb/Evolution/AppState/app-state.ts]: aritfact

File: ClientWeb/Evolution/Helpers/enum-helper.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
+import {Injectable} from "@angular/core";
+import {SiteEnum} from "../Enums/site-enum";
+
+//TODO: only for test & development purpose. MUST BE REMOVED!
+interface Map<T> {
+    [key: string]: T;
+}
+class Test{
```
Comment (tapy-lappy) [inline ClientWeb/Evolution/Helpers/enum-helper.ts]: artifact

File: ClientWeb/Evolution/Helpers/enum-helper.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
+import {Injectable} from "@angular/core";
+import {SiteEnum} from "../Enums/site-enum";
+
+//TODO: only for test & development purpose. MUST BE REMOVED!
+interface Map<T> {
```
Comment (tapy-lappy) [inline ClientWeb/Evolution/Helpers/enum-helper.ts]: artifact

File: ClientWeb/Evolution/Helpers/enum-helper.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    //     // let value: Map<number>['foo'];  //undefined
+    //
+    //     // let keys: keyof Map<number> = dict.toString(); // string
+    //     // let value: Map<number>['does not matter what will be here'] = dict["AOR"]; // number
+    //
+    //         // console.log(typeof enumValue);  //number
+    //         // console.log(EnumHelper.getNames(typeof enumValue)); //["n", "u", "m", "b", "e", "r"]
+    //     //return EnumHelper.getNames(enumValue)[0];
+    // }
+
```
Comment (tapy-lappy) [inline ClientWeb/Evolution/Helpers/enum-helper.ts]: need to clean out these comments

**Review by tapy-lappy — CHANGES_REQUESTED** [review]
Needed little changes, as removing some artifacts & cleaning one file from comments.

**Comment by AndreyAttr** [comment]
FIxed

File: ClientWeb/Evolution/Helpers/enum-helper.ts [hunk 5]
```diff
@@ -1,51 +1,8 @@
...
-    //     // let keys: keyof Map<number> = dict.toString(); // string
-    //     // let value: Map<number>['does not matter what will be here'] = dict["AOR"]; // number
-    //
-    //         // console.log(typeof enumValue);  //number
-    //         // console.log(EnumHelper.getNames(typeof enumValue)); //["n", "u", "m", "b", "e", "r"]
-    //     //return EnumHelper.getNames(enumValue)[0];
-    // }
-
-
-
```
Comment (tapy-lappy) [inline ClientWeb/Evolution/Helpers/enum-helper.ts]: excellent

**Review by tapy-lappy — APPROVED** [review]
Approved

**Comment by AndreyAttr** [comment]
**_# Merged by @AndreyAttr_**
ok

**Pull request merged**